### PR TITLE
Crimre 52 send back from review

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change

Adds form for capturing return reason. Records the Return Event.

## Link to relevant ticket

[CRIMRE-52-send-back-from-review](https://github.com/ministryofjustice/laa-review-criminal-legal-aid/tree/CRIMRE-52-send-back-from-review)

## Notes for reviewer

Blocked on reporting to the datastore proper until this PR is merged: https://github.com/ministryofjustice/laa-criminal-applications-datastore/pull/33

### After changes:
Screenshot 2023-01-24 at 15 00 13
Screenshot 2023-01-24 at 15 00 05
Screenshot 2023-01-24 at 15 03 55
Screenshot 2023-01-24 at 15 00 29

### CODE REVIEW

**Review by willmcb — APPROVED** [review]
LGTM

File: app/models/return_details.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class ReturnDetails < ApplicationStruct
+  RETURN_REASONS = Types::RETURN_REASONS
+
+  attribute? :reason, Types::Nil | Types::String
+  attribute? :details, Types::Nil | Types::String
+
+  validates :details, length: { minimum: 10 }
```
Comment (arthurashman) [inline app/models/return_details.rb:7]: I appreciate this is sensible, but did we not agree to not have any restrictions on length of text?

Comment (timpeat) [reply app/models/return_details.rb:7]: Ah perhaps I misunderstood the ticket. I thought we required some details, but would not restrict the maximum number of characters.

Comment (arthurashman) [reply app/models/return_details.rb:7]: Aha maybe that's right and I got confused between max and min!

**Pull request merged**